collect_app/src/main/java/org/odk/collect/android/database/FormDatabaseMigrator.java

```diff
@@ -45,97 +24,51 @@
-            LAST_DETECTED_FORM_VERSION_HASH};
-
-    private static final String[] COLUMN_NAMES_V9 = {_ID, DISPLAY_NAME, DESCRIPTION,
-            JR_FORM_ID, JR_VERSION, MD5_HASH, DATE, FORM_MEDIA_PATH, FORM_FILE_PATH, LANGUAGE,
-            SUBMISSION_URI, BASE64_RSA_PUBLIC_KEY, JRCACHE_FILE_PATH, AUTO_SEND, AUTO_DELETE,
-            LAST_DETECTED_FORM_VERSION_HASH, DELETED};
-
-    public static final String[] CURRENT_VERSION_COLUMN_NAMES = COLUMN_NAMES_V9;
+            "lastDetectedFormVersionHash"};
 
     // These exist in database versions 2 and 3, but not in 4...
     private static final String TEMP_FORMS_TABLE_NAME = "forms_v4";
     private static final String MODEL_VERSION = "modelVersion";
 
-    private static boolean isDatabaseBeingMigrated;
-
-    public FormsDatabaseHelper() {
-        super(new DatabaseContext(new StoragePathProvider().getDirPath(StorageSubdirectory.METADATA)), DATABASE_NAME, null, DATABASE_VERSION);
-    }
-
-    public static String getDatabasePath() {
-        return new StoragePathProvider().getDirPath(StorageSubdirectory.METADATA) + File.separator + DATABASE_NAME;
-    }
-
-    @Override
     public void onCreate(SQLiteDatabase db) {
         createFormsTableV9(db);
     }
 
     @SuppressWarnings({"checkstyle:FallThrough"})
-    @Override
-    public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {
-        try {
-            Timber.i("Upgrading database from version %d to %d", oldVersion, newVersion);
-
-            switch (oldVersion) {
-                case 1:
-                    upgradeToVersion2(db);
-                case 2:
-                case 3:
-                    upgradeToVersion4(db, oldVersion);
-                case 4:
-                    upgradeToVersion5(db);
-                case 5:
-                    upgradeToVersion6(db);
-                case 6:
-                    upgradeToVersion7(db);
-                case 7:
-                    upgradeToVersion8(db);
-                    break;
-                case 8:
-                    upgradeToVersion9(db);
-                default:
-                    Timber.i("Unknown version %s", oldVersion);
-            }
-
-            Timber.i("Upgrading database from version %d to %d completed with success.", oldVersion, newVersion);
-            isDatabaseBeingMigrated = false;
-        } catch (SQLException e) {
-            isDatabaseBeingMigrated = false;
-            throw e;
+    public void onUpgrade(SQLiteDatabase db, int oldVersion) throws SQLException {
+        switch (oldVersion) {
+            case 1:
+                upgradeToVersion2(db);
+            case 2:
+            case 3:
+                upgradeToVersion4(db, oldVersion);
+            case 4:
+                upgradeToVersion5(db);
+            case 5:
+                upgradeToVersion6(db);
+            case 6:
+                upgradeToVersion7(db);
+            case 7:
+                upgradeToVersion8(db);
+                break;
```

Comment:
Move `break` down to allow upgrades from versions other than 8.

Reply:
Oh damn. Yes. I might try and add some tests to drive this out.